src/target/adiv5.c

```diff
@@ -691,10 +691,11 @@ void adiv5_dp_init(ADIv5_DP_t *dp)
 	 *
 	 * for SWD-DP, we are guaranteed to be DP v1 or later.
 	 */
-	volatile uint32_t dpidr;
+	volatile uint32_t dpidr = idcode;
```

Comment:
I think we should not attribute idcode to dpidr, they are explicitly "different", and it was the kind of thing that initialy confused me leadind to the rework on this code,  since in the end we opted to look for a particular match idcode we dont need to either, but we should initialize it to 0